taqueria-vscode-extension/src/test/suite/extension.test.ts

```diff
@@ -5,102 +5,112 @@ import * as vscode from 'vscode';
 import { ExtensionContext, Uri } from 'vscode';
 import * as taqueriaExtension from '../../extension';
 import * as MockedObject from './MockedObject';
-import { sleep } from './utils/utils';
 
-const projectRoot = path.resolve(__dirname, '../../../');
-const testProjectDestination = `${projectRoot}/out/vscode-taq-test-project`;
-const testProjectSource = `${projectRoot}/src/test/suite/data/vscode-taq-test-project`;
-
-describe('Extension Test Suite', () => {
+const homedir = require('os').homedir();
+const sourceFilesRoot = path.resolve(__dirname, '../../../');
+const testProjectRoot = path.resolve(homedir, 'TVsCE_e2e');
+const testProjectSource = `${sourceFilesRoot}/src/test/suite/data`;
+const ligoContractFileSource = `${testProjectSource}/hello-tacos.mligo`;
+
+const testProjectDestination = `${testProjectRoot}/vscode-taq-test-project`;
+const ligoContractFileDestination = `${testProjectDestination}/contracts/hello-tacos.mligo`;
+
+const originalMethods = {
+	'window.showInformationMessage': vscode.window.showInformationMessage,
+	'window.showQuickPick': vscode.window.showQuickPick,
+};
+
+let vscodeMock: typeof vscode;
+let mockedReturnValues = {
+	openDialogMockValues: [''],
+	openDialogMockIndex: 0,
+};
+
+function setOpenDialogMocks(...mockedPaths: string[]) {
+	mockedReturnValues.openDialogMockValues = mockedPaths;
+	mockedReturnValues.openDialogMockIndex = 0;
+}
+
+function getNextOpenDialogUri() {
+	if (
+		!mockedReturnValues.openDialogMockValues
+		|| mockedReturnValues.openDialogMockIndex >= mockedReturnValues.openDialogMockValues.length
+	) {
+		throw new Error('We have already used all mocked values.');
+	}
+	const mockedValue = mockedReturnValues.openDialogMockValues[mockedReturnValues.openDialogMockIndex];
+	mockedReturnValues.openDialogMockIndex++;
+	return mockedValue;
+}
+
+let choosePlugin: string = '@taqueria/plugin-smartpy';
+
+const mockedMethods = {
+	'window.showInformationMessage': (msg: string) => Promise.resolve(console.log(msg)),
+	'window.showQuickPick': () => Promise.resolve([choosePlugin]),
+	'window.showOpenDialog': () => {
```

Comment:
Not important now, but `window.showOpenDialog` is not backed-up in `originalMethods` in line 18-21.